### ISSUE TITLE
https://learn.microsoft.com/en-us/aspnet/core/blazor/security/server/additional-scenarios?view=aspnetcore-8.0#pass-tokens-to-a-server-side-blazor-app

### ISSUE THREAD

**Issue by GStoynev** [dotnet/AspNetCore.Docs#31759]
### Description

Does the approach with capturing at initial request and passing data to components really work with server-side interactive, using the VS template/App root component? 

https://learn.microsoft.com/en-us/aspnet/core/blazor/security/server/additional-scenarios?view=aspnetcore-8.0#pass-tokens-to-a-server-side-blazor-app

@javiercn mentions it here, too https://github.com/dotnet/aspnetcore/issues/52759

The documentation example is capturing the HttpContext during root component's OnInitialize. 

OnInitialize fires only once - during pre-rendering - and stores the values in the scoped TokenProvider. 

But TokenProvider is re-created during the interactive phase and the new copy (which is what is visible in the SignalR scope) is never assigned the captured values.

In my experiment, I use a modified version of that example - one that simply captures all cookies. Like this:

**App.razor**
```
<!DOCTYPE html>
<html lang="en">

<head>   
    ...
    <HeadOutlet @rendermode="InteractiveServer" />
</head>

<body>
    <Routes @rendermode="InteractiveServer" />
    <script src="_framework/blazor.web.js"></script>
</body>

</html>

@code {
    [CascadingParameter]
    public HttpContext? HttpContext { get; set; }

    [Inject]
    TokenProvider? TokenProvider{ get; set; }

    [Inject]
    private ILogger<App> _logger { get; set; } = null!;

    protected override Task OnInitializedAsync()
    {
        using var _ = _logger.BeginScope(nameof(OnInitializedAsync));

        ArgumentNullException.ThrowIfNull(TokenProvider);

        var cookies = HttpContext?.Request.Cookies;

        _logger.LogDebug("TokenProvider Id: {TokenProvider.Id}. Cookies: {Cookies}", TokenProvider.Id, cookies);

        TokenProvider.Cookies = cookies;

        return base.OnInitializedAsync();
    }
}
```

... and that is not working as I understand the documentation example

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/blazor/security/server/additional-scenarios?view=aspnetcore-8.0#pass-tokens-to-a-server-side-blazor-app

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/security/server/additional-scenarios.md

### Document ID

c98be365-408d-7ee6-cb74-14c44d01b0b8

### Article author

@guardrex

**Comment by guardrex** [comment]
Hello @GStoynev ... This is a duplicate of https://github.com/dotnet/AspNetCore.Docs/issues/31691. I'm not sure when the product unit will review and update the approach, but I hope it will be soon. I just sent a reminder to them on Thursday of last week.
